src/App.js

```diff
@@ -1,102 +1,83 @@
-    '#FB6964',
-    '#342224',
-    '#472E32',
-    '#BDBB99',
-    '#77B1A9',
-    '#73A857'
-  ];
-const fadingEffect=()=>{
-
-  
-  let elements =document.getElementsByClassName("faded");
-
-  Array.from(elements).forEach(element=>{
-    element.classList.add("faded-text");
-  element.addEventListener("animationend",function(){
-    element.classList.remove("faded-text");
-    element.style.opacity="1";
-  });});
-}
+  const [randomNumber, setRandomNumber] = useState(0);
+  const [quotes, setQuotes] = useState(null);
+  const [singleQuote, setSingleQuote] = useState(" Life isn’t about getting and having, it’s about giving and being.");
+  const [nextIndex, setNextIndex] = useState(0);
+  const [fade, setFade] = useState(true)
 
-const getQuoteAndAuthor=()=>{
 
-  setQuote(quotes[randomNumber].quote);
-  setAuthor(quotes[randomNumber].author)
+  const generateRandomNumber = () => {
+    setFade(true)
+    setRandomNumber(Math.floor(quotes.length * Math.random()));
+    setNextIndex(nextIndex + 1 % colors.length);
+    setSingleQuote(quotes[randomNumber]);
+  }
 
-}
-const generateRandomNumber=()=>{
+  useEffect(() => {
+    fetch(QuoteUrl)
+      .then((response) => response.json())
+      .then((data) => {
+        setQuotes(data.quotes);
+        setSingleQuote(data.quotes[0])
+      });
+  }, []);
 
-  fadingEffect();
-  setRandomNumber(Math.floor(quotes.length*Math.random()) );
-  setNextIndex( ++nextIndex % colors.length);
-  getQuoteAndAuthor();
- 
-}
- 
-  
- 
-useEffect(() => {
 
-  fetch(QuoteUrl)
-    .then((response) => response.json())
-    .then((data) => {
-      
-      setQuotes(data.quotes);
-     });
-}, []);
 
+  const handleOnAnimationEnd = () => {
+    setFade(false)
+  }
 
 
- 
   return (
     <div className="App"  >
-      <header className="App-header" style={{"backgroundColor":colors[nextIndex],"transition-duration":"1s"}}>
-      <div id="quote-box">
-        {quotes &&<p id="text" class="faded" style={{"color":colors[nextIndex],"transition-duration":"1s"}} >
-            <FontAwesomeIcon icon={faQuoteLeft}></FontAwesomeIcon>  { quote }</p>}
-    
-        {quotes && <p id="author" class="faded" style={{"color":colors[nextIndex],"transition-duration":"1s"}} >
-                        -  { author}</p>}
-       
-        <div id="buttons"> 
-        <a 
-          style={{"color":colors[nextIndex],"transition-duration":"1s"}} id="tweet-quote" 
-          href={encodeURI(`http://www.twitter.com/intent/tweet?text=${quote}${author}`)} target="_blank" 
-          rel="noreferrer"> 
-            <FontAwesomeIcon id="tweet-icon" icon={faTwitterSquare}></FontAwesomeIcon> 
-        </a>
-
-         <button 
-          style={{"backgroundColor":colors[nextIndex]}} id="new-quote" onClick={()=>{generateRandomNumber()}}>
-          New Quote
+      <header className="App-header" style={{ "backgroundColor": colors[nextIndex], "transition-duration": "1s" }}>
+        <div id="quote-box" onAnimationEnd={handleOnAnimationEnd}>
```

Comment:
and i think you could just use one class here. rather 2 class on child. and see i use `onAnimationEnd={handleOnAnimationEnd}` here react way, without modifying dom directly